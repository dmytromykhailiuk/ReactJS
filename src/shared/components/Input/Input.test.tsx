import { Input } from "./";
import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

const event = {} as any;

describe("Input", () => {
  it("should match first snapshot", () => {
    const wrapper = shallow(
      <Input
        value={"1211"}
        name={"name"}
        type={"text"}
        placeholder={"write you name"}
        label={"label"}
        disabled={false}
        onChange={() => {}}
        onBlur={() => {}}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should error error message when has error", () => {
    const wrapper = shallow(
      <Input
        error={"some error"}
      />
    );   
    expect(wrapper.exists(".input__error")).toBeTruthy()
  })

  it("should call onBlur function when blur event occurred", () => {
    const onBlur = jest.fn();
    const wrapper = shallow(
      <Input
        onBlur={onBlur}
      />
    );
    wrapper.find("input").simulate("focus").simulate("blur", event);
    expect(onBlur).toHaveBeenCalledWith(event)
  })

  it("should call onChange function when change event occurred", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <Input
        onChange={onChange}
      />
    );
    wrapper.find("input").simulate("change", event);
    expect(onChange).toHaveBeenCalledWith(event)
  })
})
