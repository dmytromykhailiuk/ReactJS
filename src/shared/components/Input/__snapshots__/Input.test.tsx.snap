// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match first snapshot 1`] = `
<Fragment>
  <label
    className="label"
    htmlFor="name"
  >
    label
  </label>
  <input
    className="input"
    disabled={false}
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="write you name"
    type="text"
    value="1211"
  />
</Fragment>
`;
